pkg/minio/minio.go

```diff
@@ -160,26 +161,38 @@ func (m *Minio) IsMinioDeployed(ctx context.Context) (bool, error) {
 	return true, nil
 }
 
-// PushToMinio pushes data (i.e. a reader) to Minio
-func (m *Minio) PushToMinio(ctx context.Context, localReader io.Reader, minioFilePath, bucketName string) error {
+func (m *Minio) init(ctx context.Context) error {
+	if m.cli != nil {
+		return nil
+	}
+
 	endpoint, err := m.getEndpoint(ctx)
 	if err != nil {
 		return ErrMinioFailedToGetEndpoint.Wrap(err)
 	}
 
-	cli, err := miniogo.New(endpoint, &miniogo.Options{
+	m.cli, err = miniogo.New(endpoint, &miniogo.Options{
 		Creds:  credentials.NewStaticV4(rootUser, rootPassword, ""),
 		Secure: false,
 	})
 	if err != nil {
 		return ErrMinioFailedToInitializeClient.Wrap(err)
 	}
 
-	if err := m.createBucketIfNotExists(ctx, cli, bucketName); err != nil {
+	return nil
+}
+
+// PushToMinio pushes data (i.e. a reader) to Minio
+func (m *Minio) PushToMinio(ctx context.Context, localReader io.Reader, minioFilePath, bucketName string) error {
+	if err := m.init(ctx); err != nil {
```

Comment:
If we are refactoring this, why not refactor to actually have a `New` method?  Instead of trying to, or checking for initialization on every call of this method?

Reply:
that makes sense, we can do the check in the knuu pkg and if user wants to use it, then we deploy it and start to use it

Reply:
Will you make these updates then? That was the original scope of the issue you made anyways. 
Or do you want to do that in a follow up?

Reply:
I will do them in this PR.

Reply:
It is done, could you please have a look?